src/wazuh.py

```diff
@@ -353,3 +355,59 @@ def _generate_cluster_snippet(
             <disabled>no</disabled>
         </cluster>
     """
+
+
+def change_api_password(old_password: str, new_password: str) -> None:
+    """Change Wazuh's API password for the default 'wazuh' user.
+
+    Args:
+        old_password: the old API password for the 'wazuh' user.
+        new_password: the new API password for the 'wazuh' user.
+
+    Raises:
+        WazuhInstallationError: if an error occurs while processing the requests.
+    """
+    # The certificates might be self signed and there's no security hardening in
+    # passing them to the request since tampering with `localhost` would mean the
+    # container filesystem is compromised
+    try:
+        r = requests.get(  # nosec
+            "https://localhost:55000/security/user/authenticate",
+            auth=("wazuh", new_password),
+            timeout=10,
+            verify=False,
+        )
+        # The new password already matches. Nothing to do.
+        if r.status_code == 200:
+            return
+        # r = requests.put(  # nosec
+        #     "https://localhost:55000/security/users/2",
+        #     auth=("wazuh", old_password),
+        #     data={"password": secrets.token_hex()},
+        #     timeout=10,
+        #     verify=False,
+        # )
+        # r.raise_for_status()
+        r = requests.put(  # nosec
+            "https://localhost:55000/security/users/1",
+            auth=("wazuh", old_password),
+            data={"password": new_password},
+            timeout=10,
+            verify=False,
+        )
+        r.raise_for_status()
+    except requests.exceptions.RequestException as exc:
+        logger.error("Error modifying the default password: %s", exc)
```

Comment:
In case of a timeout error, wouldn't this log a valid password?

Reply:
The exception itself doesn't print the payload